pkg/net/url.go

```diff
@@ -74,7 +75,15 @@ func (u *URL) UnmarshalJSON(data []byte) (err error) {
 		return err
 	}
 
+	// ParseURL removes the trailing '/' in path.Clean
+	if strings.HasSuffix(s, "/") {
+		// Preserving the actual path.
+		uu, _ := url.Parse(s)
+		*ru = URL(*uu)
```

Comment:
This is wrong. you would need to fix `ParseURL()` to preserve trailing single `/`. You will also need to examine places where this package is used across minio code for breakage.

Reply:
Done :+1: